improver/cli/normalise_to_reference.py

```diff
@@ -37,37 +37,70 @@
-    corresponding data in reference.
+    """Module to enforce that the sum of data in a list of cubes is equal to the
+    corresponding data in a reference cube.
 
-    The data are updated as follows, if input cubes is a cubelist of 2 cubes with data
-    points a and b respectively and corresponding reference r, then:
+    The data are updated as follows, if there are 2 cubes to be normalised containing
+    data points a and b respectively with corresponding reference r, then:
 
     a_new = r * (a / (a + b))
     b_new = r * (b / (a + b))
 
     which ensures that r = a_new + b_new.
 
     Args:
-        cubes: (iris.cube.CubeList) cubes whose data will be updated so that the total
-            of these data is equal to the data in reference.
-        reference: (iris.cube.Cube) a cube containing the data which the data in cubes
-            will be updated with respect to.
-        ignore_zero_total: determines whether an error will be raised in the instance
-            where the total of cubes is zero but the corresponding value in reference
-            is non-zero. If True this error will not be raised - instead the data in
-            cubes will remain as zero values, if False an error will be raised.
+        cubes (List of iris.cube.Cube): A list of cubes containing both the cubes to be
+            updated and the reference cube. The reference cube will be identified by
+            matching the cube name to reference_name.
+        reference_name (str): The name of the reference cube, this should match exactly
+            one cube in cubes.
+        return_name (str): The name/s of the cubes to be returned. If None then all
+            updated cubes will be returned, otherwise only updated cubes with names
+            matching return_name will be returned. Defaults to None.
+        ignore_zero_total (bool): Determines whether an error will be raised in the
+            instance where the total of non-reference cubes is zero but the
+            corresponding value in reference is non-zero. If True this error will not be
+            raised - instead the data in the updated cubes will remain as zero values,
+            if False an error will be raised.
 
     Returns:
-        iris.cube.CubeList:
+        iris.cube.Cube or iris.cube.CubeList:
             A cubelist containing cubes with identical metadata to those in input cubes,
             but with data updated so that the total of the data in these cubes is equal
-            to the reference.
+            to the reference. If only one cube is being returned, then a cube is
+            returned instead of a cubelist.
 
     """
 
-    from improver.utilities.forecast_reference_enforcement import normalise_to_reference
+    from improver.utilities.forecast_reference_enforcement import (
+        normalise_to_reference,
+        split_cubes_by_name,
+    )
 
-    return normalise_to_reference(cubes, reference, ignore_zero_total)
+    cubes = cubes[0]
```

Comment:
This should not be here.

Reply:
I think I understand why this was put here, but it's not doing what you want when called from the command line; actually the behaviour of the acceptance tests is rather weird. From the command line, calling this CLI with multiple cubes, the cubes variable becomes something like this:

```
(
    [
        <iris 'Cube' of probability_of_lwe_thickness_of_precipitation_amount_above_threshold / (1) (lwe_thickness_of_precipitation_amount: 27; projection_y_coordinate: 970; projection_x_coordinate: 1042)>
    ],
    [
        <iris 'Cube' of probability_of_thickness_of_rainfall_amount_above_threshold / (1) (thickness_of_rainfall_amount: 27; projection_y_coordinate: 970; projection_x_coordinate: 1042)>
    ],
    [
        <iris 'Cube' of probability_of_lwe_thickness_of_sleetfall_amount_above_threshold / (1) (lwe_thickness_of_sleetfall_amount: 27; projection_y_coordinate: 970; projection_x_coordinate: 1042)>
    ],
    [
        <iris 'Cube' of probability_of_lwe_thickness_of_snowfall_amount_above_threshold / (1) (lwe_thickness_of_snowfall_amount: 27; projection_y_coordinate: 970; projection_x_coordinate: 1042)>
    ]
)
```

Meaning that your cubes = cubes[0] line just gives you:
```
cubes = [
    <iris 'Cube' of probability_of_lwe_thickness_of_precipitation_amount_above_threshold / (1) (lwe_thickness_of_precipitation_amount: 27; projection_y_coordinate: 970; projection_x_coordinate: 1042)>
],
```

However, your acceptance tests create a single list wrapped in a tuple:

```
(
    [
        <iris 'Cube' of probability_of_lwe_thickness_of_precipitation_amount_above_threshold / (1) (lwe_thickness_of_precipitation_amount: 27; projection_y_coordinate: 970; projection_x_coordinate: 1042)>,
        <iris 'Cube' of probability_of_thickness_of_rainfall_amount_above_threshold / (1) (thickness_of_rainfall_amount: 27; projection_y_coordinate: 970; projection_x_coordinate: 1042)>,
        <iris 'Cube' of probability_of_lwe_thickness_of_sleetfall_amount_above_threshold / (1) (lwe_thickness_of_sleetfall_amount: 27; projection_y_coordinate: 970; projection_x_coordinate: 1042)>,
        <iris 'Cube' of probability_of_lwe_thickness_of_snowfall_amount_above_threshold / (1) (lwe_thickness_of_snowfall_amount: 27; projection_y_coordinate: 970; projection_x_coordinate: 1042)>
    ]
)
```
Hence your command yields a simple list that can be iterated over.
I have absolutely no idea why calling the CLI from the acceptance test is behaving differently to the command line and it perturbs me. However, the solution is to use the `flatten` utility which is used in some other CLIs that accept a cubelist. This will enable the acceptance tests and command line calls to behave in the same way.

```
from improver.utilities.flatten import flatten

cubes = flatten(cubes)
```